File: framework/core/lib/kfs.js
#!/usr/bin/env node

const path = require('path');
const { kfs } = require('./executable');
const shell = require('./shell');

shell.run(kfs, process.argv.slice(2), true, {
  silent: true,
  env: {
    KFS_PATH: path.join(
      require.resolve('@kungfu-tech/sdk'),
      '..',
      '..',
      'dist',
      'sdk',
      'kfs.js',
    ),
    KUNGFU_PATH: path.join(__dirname, '..', 'dist', 'kungfu'),
  },
});

File: framework/core/.devtools/kfs.py
#  SPDX-License-Identifier: Apache-2.0

from env import __frozen__, kungfu_dir
from kungfu import __tool__ as origin
from os import environ
from pathlib import Path

environ["KFS_PATH"] = Path("../../developer/sdk/dist/sdk/kfs.js").resolve().as_posix()
environ["KUNGFU_PATH"] = kungfu_dir.replace("\\", "/")

if __frozen__:
    origin.sdk()
